style: spell out constants used in crypto interfaces

ppannuto opened the pull request:
### Pull Request Overview

This is inspired by [the impending addition of yet one more `_L` constant](https://github.com/tock/tock/pull/4395#discussion_r2074661289). I was going to suggest writing things out as a clearer alternative on that PR, but realized it was really just following the existing style.

While it is generally [*] convention to use single letters for trait bounds, the same isn't quite as universal for const generics. Our [Style Guide](doc/Style.md#using-descriptive-names) prefers descriptive names. This feels like a use case where there isn't huge overhead in physical space on the screen from writing these out, and there is much more clarity in reading standalone use, especially as uses get further from the defining/explaining site in the HIL. I went `_LEN` in favor of `_LENGTH` following a general preference for `len` in Rust.

[*] In my ever-so-humble-opinion, _a terrible convention_

### Testing Strategy

This pull request was tested by compiling.


### TODO or Help Wanted

Vibes check? Am I just being curmudgeonly about a Rust-ism with single-letter nonsense, or do others also feel this improves the readability? [Again, _especially_ with an expectation of yet more `_L` constants coming]

### Documentation Updated

- [x] Updated the relevant files in `/docs`, or no updates are required.

### Formatting

- [ ] Ran `make prepush`.

bradjc commented:
I personally don't mind the `H` + `HL` convention. I do think `HASH_LEN` is more clear. We also need to decide what to do in the digest and entropy HILs.

ppannuto commented:
Here's what things look like if you expand `H` to `HashKind` and `S` to `SignatureKind` as well: https://github.com/tock/tock/compare/write-out-crypto-types-too?expand=1

I do favor that, though I cede the point that it is more verbose

lschuermann reviewed (dismissed):
I think this is a solid improvement.

> While it is generally [*] convention to use single letters for trait bounds

I believe I've seen a fair share of examples where trait bounds were camel-case identifiers, esp. where there is ambiguity. In the diff most trait implementations or type definitions that contain generic trait bounds are split over multiple lines anyways, so I'd be totally happy to see these be expanded as well.

But this alone is a good improvement in on itself, and I don't feel particularly strongly about the single-letter trait bounds.

lschuermann commented on boards/components/src/appid/checker_signature.rs:
```diff
@@ -10,50 +10,63 @@ use kernel::hil::{digest, public_key_crypto};
-pub type AppCheckerSignatureComponentType<S, H, const HL: usize, const SL: usize> =
-    capsules_system::process_checker::signature::AppCheckerSignature<'static, S, H, HL, SL>;
+pub type AppCheckerSignatureComponentType<
+    SignatureKind,
+    HashKind,
```
Might be nitpicky, but these fields don't really refer to a `SignatureKind` "marker" as much as they refer to a `Signer` "component" that actually does work, right? Same for `HashKind` -> `Hasher`?

I'm slightly confused by the `Kind` part in the generic's name, but it's not a big deal either way.

ppannuto replied:
This is a good nitpick to discuss, the whole point is clarity from more explicit names.

This is also definitely a case of Pat might be learning Rust via PR discussions... but my rationale for the current naming was that, IIUC, this is ultimately the placeholder label for a type. E.g., looking at [the documentation explaining bounds](https://doc.rust-lang.org/rust-by-example/generics/bounds.html) this would be a "generic type `SignatureKind`". I used "Kind" in lieu of "Type" as I thought that was a common convention for avoiding complexity around "Type" as a keyword, though maybe it is not needed here.

Would `SignerType` (and `HasherType`) make more sense?

bradjc reviewed (changes requested):
I understand spelling out the constants (per the title) but why are the type names changing?

lschuermann commented:
I don't think any type names are changing, just constants and generics placeholders.

But yes, this now goes beyond constants, and thus the PR title / description no longer accurately matches this changeset.

bradjc commented:
> I don't think any type names are changing, just constants and generics placeholders.

I guess I'm referring to the generics placeholders, which I don't think we should start changing now.

lschuermann commented:
Do you object to these changes at all, or just as part of this PR mixed in with the other changes?

bradjc commented:
> Do you object to these changes at all, or just as part of this PR mixed in with the other changes?

Entirely; I thought we agreed that for better or worse that is Rust convention, and we use it pervasively throughout the kernel such that changing in one place just adds confusion and inconsistency.

bradjc commented:
> I think I was mostly irritated by the single letter type parameters in the macro definition

macros are generally bad

ppannuto commented:
I updated the whole thing to do both because after my [comment with exemplar expanding everything](https://github.com/tock/tock/pull/4443#issuecomment-2898542341) there was one agreeing comment and then silence. I [did say I updated both](https://github.com/tock/tock/pull/4443#issuecomment-2978298651), but missed the PR title, sorry (somehow, my brain just always glazes over PR titles; I think I forget they are editable; I just use the comment thread as the current state of truth when reading PR updates. I fully acknowledge that is a weird thing to do and updating titles is probably useful for most folks).

---

On the merits of the question at hand, [Rust naming convention](https://rust-lang.github.io/api-guidelines/naming.html) for Type Parameters is `concise UpperCamelCase, usually single uppercase letter: T`.

Digging around some, it seems that the thread Leon link captures the pseudo-convention that seems to be the norm across Rust, Java, and C++ from current search results, i.e., single letter names for things that are _truly_ generic types (`Box<T>`) and concise identifiers for generic types with some semantic restrictions (e.g., [`<Idx>` in Rust core](https://doc.rust-lang.org/src/core/range.rs.html#124) was an example linked from one discussion).

What do we think about `H --> Hsh` and `S --> Sig`?

bradjc commented:
I remember a conversation somewhere where we said the convention in both Rust and Tock is single letters for types but not for constants. I thought that was the understanding going in to this PR.

We use single letter types everywhere in Tock, I don't see why this is the PR to change that convention. I think `H` and `S` remain clear, particularly with the full description always within 5 lines of code. Plus, using single letter for types makes it clear that there is no actual type called `Sig` or `Aes` or `Alm` (for example) that the reader should look for. The _only_ time we use single letters is with these generic types, so it is clear what they are and where they are defined. In the case we have name conflicts I think we use two upper case letters, again providing the same distinction.
